lib/ansible/modules/packaging/os/apt.py

```diff
@@ -753,6 +753,12 @@ def upgrade(m, mode="yes", force=False, default_release=None,
     else:
         force_yes = ''
 
+    if apt_cmd is None:
```

Comment:
I don't see how this will ever evaluate to `True`. The `if` statement on line 730 should always populate `apt_cmd`.

Reply:
when name is given '*' and state as latest it triggers 'upgrade' with mode as 'yes' (line 1003). This is seen only when APTITUDE is not installed on given machine.

Reply:
Ah, that'll do it. :+1: